subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/ProjectLoadingIntegrationTest.java

```diff
@@ -217,9 +218,48 @@ public void ignoresMultiProjectBuildInParentDirectoryWhichDoesNotMeetDefaultProj
                 "}");
         testFile("subdirectory/gradle.properties").write("prop=value");
 
-        inDirectory(subDirectory).withSearchUpwards().withTasks("do-stuff").run();
-        usingProjectDir(subDirectory).withSearchUpwards().withTasks("do-stuff").run();
-        usingBuildFile(buildFile).withSearchUpwards().withTasks("do-stuff").run();
+        inDirectory(subDirectory).withSearchUpwards().withTasks("do-stuff").expectDeprecationWarning().run();
+        usingProjectDir(subDirectory).withSearchUpwards().withTasks("do-stuff").expectDeprecationWarning().run();
+        usingBuildFile(buildFile).withSearchUpwards().withTasks("do-stuff").expectDeprecationWarning().run();
+    }
+
+    @Test
+    public void deprecationWarningAppearsWhenNestedBuildHasNoSettingsFile() {
+        testFile("settings.gradle").write("include 'another'");
+
+        TestFile subDirectory = getTestDirectory().file("sub");
+        TestFile subBuildFile = subDirectory.file("sub.gradle").write("");
+        subDirectory.file("build.gradle").write("");
+
+        ExecutionResult result = inDirectory(subDirectory).withSearchUpwards().withTasks("help").expectDeprecationWarning().run();
+        result.assertOutputContains("Support for nested build without a settings file was deprecated and will be removed in Gradle 5.0. You should create a empty settings file in " + subDirectory.getAbsolutePath());
+
+        result = usingBuildFile(subBuildFile).inDirectory(subDirectory).withSearchUpwards().withTasks("help").expectDeprecationWarning().run();
+        result.assertOutputContains("Support for nested build without a settings file was deprecated and will be removed in Gradle 5.0. You should create a empty settings file in " + subDirectory.getAbsolutePath());
+
+        result = usingProjectDir(subDirectory).withSearchUpwards().withTasks("help").expectDeprecationWarning().run();
+        result.assertOutputContains("Support for nested build without a settings file was deprecated and will be removed in Gradle 5.0. You should create a empty settings file in " + subDirectory.getAbsolutePath());
+    }
+
+    @Test
+    public void noDeprecationWarningAppearsWhenNestedBuildHasSettingsFile() {
+        testFile("settings.gradle").write("include 'another'");
+
+        TestFile subDirectory = getTestDirectory().file("sub");
+        TestFile subSettingsFile = subDirectory.file("renamed_settings.gradle").write("");
```

Comment:
:o: it seems that this test mixes 2 different things: having a `settings.gradle` in a sub-build, and using a different `settings.gradle` file. Shouldn't we test the 2 separately?

Reply:
Sure.